pkg/ddevapp/symfony.go

```diff
@@ -0,0 +1,122 @@
+package ddevapp
+
+import (
+	"fmt"
+	"path/filepath"
+	"slices"
+
+	"github.com/ddev/ddev/pkg/fileutil"
+	"github.com/ddev/ddev/pkg/nodeps"
+)
+
+// isSymfonyApp returns true if the app is of type symfony
+func isSymfonyApp(app *DdevApp) bool {
+	return fileutil.FileExists(filepath.Join(app.AppRoot, "bin", "console")) && fileutil.FileExists(filepath.Join(app.AppRoot, "src", "Kernel.php"))
+}
+
+// setSymfonySiteSettingsPaths sets the paths to .env.local file.
+func setSymfonySiteSettingsPaths(app *DdevApp) {
+	app.SiteSettingsPath = filepath.Join(app.AppRoot, ".env.local")
+}
+
+func symfonyEnvMailer(app *DdevApp, envMap map[string]string) {
+	envMap["MAILER_AUTH_MODE"] = ""
+	envMap["MAILER_PASSWORD"] = ""
+	envMap["MAILER_USERNAME"] = ""
+	envMap["MAILER_CATCHER"] = "1"
+	envMap["MAILER_DRIVER"] = "smtp"
+	envMap["MAILER_DSN"] = "smtp://localhost:1025"
+	envMap["MAILER_HOST"] = "localhost"
+	envMap["MAILER_PORT"] = "1025"
+	envMap["MAILER_URL"] = "smtp://localhost:1025"
+	mailpitPort := app.GetMailpitHTTPSPort()
+	if app.CanUseHTTPOnly() {
+		mailpitPort = app.GetMailpitHTTPPort()
+	}
+	envMap["MAILER_WEB_URL"] = fmt.Sprintf("%s:%s", app.GetPrimaryURL(), mailpitPort)
+}
+
+func symfonyEnvDatabase(app *DdevApp, envMap map[string]string) {
+	if slices.Contains(app.OmitContainers, "db") {
+		return
+	}
+
+	dbPort := ""
+	dbDriver := ""
+	dbVersion := ""
+
+	if app.Database.Type == nodeps.Postgres {
+		dbPort = "5432"
+		dbDriver = "postgres"
+		dbVersion = app.Database.Version
+	} else if app.Database.Type == nodeps.MariaDB || app.Database.Type == nodeps.MySQL {
+		dbPort = "3306"
+		dbDriver = "mysql"
+		if app.Database.Type == nodeps.MariaDB {
+			// doctrine requires mariadb version until its patch version so add 0 as default patch version
+			dbVersion = fmt.Sprintf("%s.0-mariadb", app.Database.Version)
```

Comment:
Can you add a link and a more detailed explanation for this please in the comment? This seems odd. I'm pretty sure Drupal uses symfony access to mariadb without this, but I could be wrong.  This seems very fragile and artificial.

Reply:
actually from Symfony docs should contain full database version https://symfony.com/doc/current/reference/configuration/doctrine.html#doctrine-dbal-configuration like from `mysql -V`

Note: I'm not using mariadb myself, if this is too fragile, I can remove it. i did that hacks because it was failing if I initialize symfony with default ddev config, so I debug the issue

only mariadb requires until its patch version https://github.com/doctrine/dbal/blob/4.2.x/src/Driver/AbstractMySQLDriver.php#L92-L108

while [postgres ](https://github.com/doctrine/dbal/blob/4.2.x/src/Driver/AbstractPostgreSQLDriver.php#L28-L33)only need the major version or no regex check on MySQL version https://github.com/doctrine/dbal/blob/4.2.x/src/Driver/AbstractMySQLDriver.php#L62-L76

Reply:
If you could take a look at it and verify that it's really required it would be great. I'm betting it's not required. It seems very awkward and error-prone to artificially add a ".0" there.

Reply:
the ".0" is not required if we can get the full mariadb version string

or what if we exclude mariadb? if any mariadb user need this in the future, they can contribute on this

Reply:
You can get the full mariadb version string but only after starting it up.

MariaDB is probably the most popular database for the majority of CMSs. This is certainly an interesting problem. 

One thing we *could* do that we've never had to deal with before is to detect the mariadb version during startup. It's easy in a post-start hook, 

`ddev exec 'mysql -e "SELECT VERSION();"'`

Reply:
Quick search on drupal repository and found this https://github.com/drupal/drupal/blob/11.x/core/modules/mysql/src/Driver/Database/mysql/Connection.php#L277
Drupal do check the db version

One thing i can do here is to exclude mariadb from auto-populated env var

Reply:
So Drupal does it dynamically, but Symfony seems to rely on it being preconfigured. 

I don't think this is going to affect mariadb only. Doesn't it affect mysql? 

I just don't understand how Symfony wouldn't be able to handle a db server upgrade, for example. A new version of DDEV often has a new patch version of mariadb, mysql, and postgres.

Reply:
Commited to exclude mariadb